### PR TITLE
Update support for stack to ghc8.10

### PR DESCRIPTION
**Pull request by molysgaard**
Usage instructions:
To use stack for building, first you need to install stack.

Instructions can be found here: https://docs.haskellstack.org/en/stable/README/

Once you have `stack` as a command in your shell, you can run

```
stack build
```

from the root folder of `jhc-components` and stack will install the
correct ghc version, find all the required packages in a single set that
has no conflicts, download and build them. Lastly stack builds
jhc-components and its packages.

To run jhc, the syntax is:
```
stack exec jhc -- my jhc args
```

The key reason this works is that in the `stack.yaml` file, there is a
reference to: `resolver: nightly-2020-08-16`. This means that stack will
query stackage.org for a non-conflicting package set that it compiled
that night. This package set is the last one compiled with ghc-8.10.

In other words, we are outsourcing the work of finding a non-conflicting
set of packages to stackage, which has a much more elaborate way of
checking this, by actually building all the packages together, than
cabal, which just relies on version and dependency information.

### CODE REVIEW

**Comment by yobson** [comment]
Good stuff!
